src/extensions/dataarray.js

```diff
@@ -0,0 +1,82 @@
+import { datastream } from '../constants';
+import express        from 'express';
+import response       from '../response';
+import * as ERR       from '../errors';
+
+/**
+ * Implementation of 13.2 "Create Observation entities with dataArray"
+ *
+ * Similar to the SWE DataArray in the OGC SOS, SensorThings API also
+ * provides the support of dataArray (in addition to formatting every
+ * observation entity as a JSON object) to aggregate multiple Observation
+ * entities and reduce the request (e.g., POST) and response (e.g., GET)
+ * size. SensorThings mainly use dataArray in two scenarios: (1) get
+ * Observation entities in dataArray, and (2) create Observation entities
+ * with dataArray.
+ *
+ **/
+
+export default version => {
+  return (req, res, next) => {
+
+    // Required properties in each object:
+    //   - Datastream
+    //   - phenomenonTime
+    //   - result
+    //   - featureOfInterest
+    let requiredProperties = ['phenomenonTime', 'result', 'FeatureOfInterest'];
+    var validateObject = function(obj) {
+      if (obj.Datastream === undefined) {
+        if (missingProperties.indexOf(datastream) === -1) {
```

Comment:
And in any case, I wouldn't use an Array for `missingProperties`. I would use an Object, set the property whenever is missing with a dummy value like `true` and do a `Object.keys(missingProperties)` to get the array of missing properties. This way the conditions are simpler:

```javascript
  if (!obj[datastream]) {
    missingProperties[datastream] = true;
  }
```